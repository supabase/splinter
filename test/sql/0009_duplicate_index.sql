begin;
  set local search_path = '';

  create table public.blog(
    id int primary key
  );

  create index ix_1 on public.blog (id);

  -- 0 issue
  select * from lint."0009_duplicate_index";

  create index ix_2 on public.blog (id);

  -- 1 issue
  select * from lint."0009_duplicate_index";

  create index ix_3 on public.blog (id);

  -- 1 issue
  select * from lint."0009_duplicate_index";

  drop index public.ix_3;
  drop index public.ix_2;
  select * from lint."0009_duplicate_index";


rollback;
