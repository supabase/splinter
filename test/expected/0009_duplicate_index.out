begin;
  set local search_path = '';
  create table public.blog(
    id int primary key
  );
  create index ix_1 on public.blog (id);
  -- 0 issue
  select * from lint."0009_duplicate_index";
 name | title | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

  create index ix_2 on public.blog (id);
  -- 1 issue
  select * from lint."0009_duplicate_index";
      name       |      title      | level |  facing  |  categories   |                        description                        |                                        detail                                        |                                     remediation                                     |                                      metadata                                      |                cache_key                
-----------------+-----------------+-------+----------+---------------+-----------------------------------------------------------+--------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------+------------------------------------------------------------------------------------+-----------------------------------------
 duplicate_index | Duplicate Index | WARN  | EXTERNAL | {PERFORMANCE} | Detects cases where two ore more identical indexes exist. | Table \`public.blog\` has identical indexes {ix_1,ix_2}. Drop all except one of them | https://supabase.com/docs/guides/database/database-linter?lint=0009_duplicate_index | {"name": "blog", "type": "table", "schema": "public", "indexes": ["ix_1", "ix_2"]} | duplicate_index_public_blog_{ix_1,ix_2}
(1 row)

  create index ix_3 on public.blog (id);
  -- 1 issue
  select * from lint."0009_duplicate_index";
      name       |      title      | level |  facing  |  categories   |                        description                        |                                          detail                                           |                                     remediation                                     |                                          metadata                                          |                  cache_key                   
-----------------+-----------------+-------+----------+---------------+-----------------------------------------------------------+-------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------+----------------------------------------------
 duplicate_index | Duplicate Index | WARN  | EXTERNAL | {PERFORMANCE} | Detects cases where two ore more identical indexes exist. | Table \`public.blog\` has identical indexes {ix_1,ix_2,ix_3}. Drop all except one of them | https://supabase.com/docs/guides/database/database-linter?lint=0009_duplicate_index | {"name": "blog", "type": "table", "schema": "public", "indexes": ["ix_1", "ix_2", "ix_3"]} | duplicate_index_public_blog_{ix_1,ix_2,ix_3}
(1 row)

  drop index public.ix_3;
  drop index public.ix_2;
  select * from lint."0009_duplicate_index";
 name | title | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

rollback;
